newdust/scatteringmodel/make_ggadt.py

```diff
@@ -0,0 +1,222 @@
+
+    overwrite: bool: whether or not to overrite a preexisting FITS file with the same name
+
+    Returns FITS file with the data described above
+    """
+
+    #data to store in FITS
+    radii = []
+    evs = []
+    theta = [] #NEED TO IMPLEMENT
+    qext = []
+    qabs = []
+    qsca = []
+    diff = [] #NEED TO IMPLEMENT
+
+    axis_ratio = 1.0
+    have_evs_and_ratio = False
+
+    #go through the output files and popuate above variables
+    i = 0
+    while i <= last_index:
+        filename= f'{folder}/{material}_{i}_{shape}_{orientation}.out'
+
+        #check file existence -- for larger astrodust models there many not be GGADT output files with certain names (ex: no random oblates from indicies 0 - 10)
+        if not os.path.isfile(filename):
+            i += 1
+            continue
+        
+        #evs and axis ratio are constant through a FITS file so they only need to be collected once
+        if have_evs_and_ratio:
+
+            data = parse_file(filename, orientation)
+            qext.append(data['qext'])
+            qabs.append(data['qabs'])
+            qsca.append(data['qsca'])
+            radii.append(data['radius'])
+
+        else:
+
+            data = parse_file(filename, orientation, have_evs_and_ratio=False)
+            qext.append(data['qext'])
+            qabs.append(data['qabs'])
+            qsca.append(data['qsca'])
+            radii.append(data['radius'])
+            evs = data['evs']
+            axis_ratio = data['axis ratio']
+
+            have_evs_and_ratio = True
+
+        i += 1
+    
+    #make parameters and header
+    header = make_header(material, shape, axis_ratio, orientation)
+    pars = make_pars(evs, radii, theta)
+    
+    img_list = []
+    for (val, head) in zip([qext, qabs, qsca, diff],
+                           ["Qext", "Qabs", "Qsca", "Diff-xsect (ster^-1)"]):
+        htemp = fits.Header()
+        htemp['TYPE'] = head
+        img_list.append(fits.ImageHDU(val, header=htemp))
+
+    #write table
+    fnl_list = [header] + pars + img_list
+    hdu_list = fits.HDUList(hdus=fnl_list)
+    hdu_list.writeto(outfile, overwrite=overwrite)
+    return outfile
+
+#Helper functions:
+
+def parse_file(filename, orientation, have_evs_and_ratio=True):
+    """
+    Parses through a GGADT output file and returns header and extinction data
+
+    filename: string: full path, including name, to a GGADT output file
+
+    orientation: string: Either 'set' or 'rand' -- 'set' denotes a set orientation (0º, 0º, 0º), 'rand' denotes a random orientation -- but the code should work as long as 'rand' denotes the random orientation (i.e. There could be multiple set orientations). This should be inputted as it appears in the file names.
+
+    have_evs_and_ratio: bool: if true, parse_file() not will record the axis ratio nor incident energy values of the grain in the given output file -- these should be the same for every output file so they only need to be recorded once
```

Comment:
Remove this keyword -- always return energy and axis ratio. The assumption that they are all the same will be checked later.